streams/src/test/java/org/apache/kafka/streams/integration/FanoutIntegrationTest.java

```diff
@@ -0,0 +1,165 @@
+ *                                         |
+ * Kafka topic A ---stream()--> stream1 ---+
+ *                                         |
+ *                                         +---map()---> stream3 ---to()---> Kafka topic C
+ *
+ * }
+ * </pre>
+ */
+public class FanoutIntegrationTest {
+    @ClassRule
+    public static EmbeddedSingleNodeKafkaCluster cluster = new EmbeddedSingleNodeKafkaCluster();
+    private static final String INPUT_TOPIC_A = "A";
+    private static final String OUTPUT_TOPIC_B = "B";
+    private static final String OUTPUT_TOPIC_C = "C";
+
+    @BeforeClass
+    public static void startKafkaCluster() throws Exception {
+        cluster.createTopic(INPUT_TOPIC_A);
+        cluster.createTopic(OUTPUT_TOPIC_B);
+        cluster.createTopic(OUTPUT_TOPIC_C);
+    }
+
+    @Test
+    public void shouldFanoutTheInput() throws Exception {
+        List<String> inputValues = Arrays.asList("Hello", "World");
+        List<String> expectedValuesForB = new ArrayList<>();
+        List<String> expectedValuesForC = new ArrayList<>();
+        for (String input : inputValues) {
+            expectedValuesForB.add(input.toUpperCase(Locale.getDefault()));
+            expectedValuesForC.add(input.toLowerCase(Locale.getDefault()));
+        }
+
+        //
+        // Step 1: Configure and start the processor topology.
+        //
+        KStreamBuilder builder = new KStreamBuilder();
+
+        Properties streamsConfiguration = new Properties();
+        streamsConfiguration.put(StreamsConfig.APPLICATION_ID_CONFIG, "fanout-integration-test");
+        streamsConfiguration.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, cluster.bootstrapServers());
+        streamsConfiguration.put(StreamsConfig.ZOOKEEPER_CONNECT_CONFIG, cluster.zKConnectString());
+        streamsConfiguration.put(StreamsConfig.VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass().getName());
+
+        KStream<byte[], String> stream1 = builder.stream(INPUT_TOPIC_A);
+        KStream<byte[], String> stream2 = stream1.mapValues(
+            new ValueMapper<String, String>() {
+                @Override
+                public String apply(String value) {
+                    return value.toUpperCase(Locale.getDefault());
+                }
+            });
+        KStream<byte[], String> stream3 = stream1.mapValues(
+            new ValueMapper<String, String>() {
+                @Override
+                public String apply(String value) {
+                    return value.toLowerCase(Locale.getDefault());
+                }
+            });
+        stream2.to(OUTPUT_TOPIC_B);
+        stream3.to(OUTPUT_TOPIC_C);
+
+        KafkaStreams streams = new KafkaStreams(builder, streamsConfiguration);
+        streams.start();
+
+        // Wait briefly for the topology to be fully up and running (otherwise it might miss some or all
+        // of the input data we produce below).
+        Thread.sleep(5000);
```

Comment:
Thanks @ijuma . We don't yet have a way to check if streams have been initialised, but are looking to add one (https://issues.apache.org/jira/browse/KAFKA-3637)